pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -162,6 +165,24 @@ public class ClientConfigurationData implements Serializable, Cloneable {
     )
     private String tlsTrustCertsFilePath = null;
 
+    @ApiModelProperty(
+            name = "tlsPrivateKey",
+            value = "The private key for the client certificate. Only used when tlsTrustStoreStream is non-null."
+    )
+    private PrivateKey tlsPrivateKey = null;
+
+    @ApiModelProperty(
+            name = "tlsCertificates",
+            value = "Client certificate chain. Only used when tlsTrustStoreStream is non-null."
+    )
+    private Certificate[] tlsCertificates = null;
+
+    @ApiModelProperty(
+            name = "tlsTrustStoreStream",
+            value = "Input-stream of the trust store. When configured, the tlsPrivateKey and tlsCertificates are used."
+    )
+    private InputStream tlsTrustStoreStream = null;
```

Comment:
please explain the lifecycle of this Stream.
is the Pulsar client intended to always read it fully and then close it ?
what happens in case of failure of the creation of the PulsarClient ?

also, who is responsible for closing it ?
as this Stream is bound to some external resource (a file, a network connection) the lifecycle of the object must be well-know.

Reply:
I think it should have been `ByteArrayInputStream`, though I'm not sure that is the right direction.